File: legacy-runtime/src/main/java/com/xiaomi/clientreport/data/PerfClientReport.java
package com.xiaomi.clientreport.data;

import com.xiaomi.channel.commonutils.logger.MyLog;
import org.json.JSONException;
import org.json.JSONObject;

/* JADX INFO: loaded from: miuipushsdkshared_3_7_9.jar:com/xiaomi/clientreport/data/PerfClientReport.class */
public class PerfClientReport extends BaseClientReport {
    private static final long DEFAULT_VALUE = -1;
    public int code;
    public long perfCounts = DEFAULT_VALUE;
    public long perfLatencies = DEFAULT_VALUE;

    public static PerfClientReport getBlankInstance() {
        return new PerfClientReport();
    }

    @Override // com.xiaomi.clientreport.data.BaseClientReport
    public JSONObject toJson() {
        try {
            JSONObject json = super.toJson();
            if (json == null) {
                return null;
            }
            json.put("code", this.code);
            json.put("perfCounts", this.perfCounts);
            json.put("perfLatencies", this.perfLatencies);
            return json;
        } catch (JSONException e) {
            MyLog.e(e);
            return null;
        }
    }

    @Override // com.xiaomi.clientreport.data.BaseClientReport
    public String toJsonString() {
        return super.toJsonString();
    }
}

File: legacy/src/main/java/com/xiaomi/clientreport/data/PerfClientReport.kt
package com.xiaomi.clientreport.data
import io.github.magisk317.mipush.protocol.model.*

import com.xiaomi.channel.commonutils.logger.MyLog
import org.json.JSONException
import org.json.JSONObject

class PerfClientReport : BaseClientReport() {
    @JvmField var code: Int = 0
    @JvmField var perfCounts: Long = DEFAULT_VALUE
    @JvmField var perfLatencies: Long = DEFAULT_VALUE

    companion object {
        private const val DEFAULT_VALUE: Long = -1

        @JvmStatic
        fun getBlankInstance(): PerfClientReport = PerfClientReport()
    }

    override fun toJson(): JSONObject? {
        return try {
            val json = super.toJson() ?: return null
            json.put("code", code)
            json.put("perfCounts", perfCounts)
            json.put("perfLatencies", perfLatencies)
            json
        } catch (e: JSONException) {
            MyLog.e(e)
            null
        }
    }

    override fun toJsonString(): String = super.toJsonString()
}
